### ISSUE TITLE
focal_loss

### ISSUE THREAD

**Issue by dl8207531** [xggIoU/centernet_tensorflow_wilderface_voc#18]
loss.py中focal_loss函数

 if num_pos == 0.0:
    loss = loss - neg_loss
  else:
    loss = loss - (pos_loss + neg_loss) / num_pos
这样写，你使用的是tf2.0用的动态图？

**Comment by xggIoU** [comment]
静态图改用tf.cond吧

**Comment by dl8207531** [comment]
再问下每张样本都有centerpoint 所以num_pos应该不会为0吧？

**Comment by dl8207531** [comment]
还有点不明白请教一下，就是没找到bbox size标签归一化的地方 还是不归一直接收敛？

**Comment by xggIoU** [comment]
训练集如果都有object 那num_pos不会为0，bbox size是直接回归的原尺寸，原论文就是这么干的。

**Comment by dl8207531** [comment]
bbox size是基于heatmap的size ，输出的就是heatmap想想也合理，只是训练的时候不要溢出了
之前不理解为什么size loss 要x0.1这么小 这样看就讲通了
